src/basic/terminal-util.c

```diff
@@ -1501,11 +1501,12 @@ int vt_restore(int fd) {
 
         assert(fd >= 0);
 
-        if (!isatty_safe(fd))
-                return log_debug_errno(errno, "Asked to restore the VT for an fd that does not refer to a terminal: %m");
```

Comment:
this should be kept in place. ioctl() numbers on Linux are assigned multiple times, hence we should not willy-nilly issue them unless we know for sure a device is of a certain type.

isatty() is the official way to detect if something is a tty (even if it internally just issues a ioctl too, but that's libc's fault then), hence let's use that here to ensure this is the right ghing.

Reply:
Ah, thanks for the clarification. A proper fix would then be to call `TCGETS` directly, instead of through the `isatty()` wrapper.